File: Domain/Affilie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public partial class Affilie
    {
        public Affilie()
        {
            AvanceCheques = new HashSet<AvanceCheque>();
            Qps = new HashSet<Qp>();
            QpMois = new HashSet<QpMois>();
            CumuleQps=new HashSet<CumuleQp>();
            MisAjours=new HashSet<MisAjour>();
            Conjoints=new HashSet<Conjoint>();
            Enfants=new HashSet<Enfant>();
            Cartes=new HashSet<Carte>();
        }

        public string Matricule { get; set; }
        public string NPension { get; set; }
        public string Nom { get; set; }
        public string Prenom { get; set; }
        [Key, Column(Order = 1)] 
        public string Cin { get; set; }
        public DateTime DateNaissance { get; set; }
        public string Telephone { get; set; }
        public string Telephone2 { get; set; }
        public string Sexe { get; set; }
        public string Adresse { get; set; }
        public string Ville { get; set; }
        public string CodePostale { get; set; }
        public string Rib { get; set; }
        public int? Rang { get; set; }
        public int? SituationVital { get; set; }
        public string TypeAffilie { get; set; }  
        public string NumAdherent { get; set; }
        public string Email { get; set; }

        public virtual ICollection<AvanceCheque> AvanceCheques { get; set; }
        public virtual ICollection<Qp> Qps { get; set; }
        public virtual ICollection<QpMois> QpMois { get; set; }
        public virtual ICollection<CumuleQp> CumuleQps { get; set; }
         public virtual ICollection<MisAjour> MisAjours { get; set; }
          public virtual ICollection<Conjoint> Conjoints { get; set; }
          
          public virtual ICollection<Enfant> Enfants { get; set; }
          public virtual ICollection<Carte> Cartes { get; set; }

    }
}

File: client-app/src/app/models/affilie.ts
import { number, string } from "prop-types"

export interface IAffiliesEnvelope {
    affilies: IAffilie[];
    affilieCount : number;
  }
  
  export interface IAffilie 
    {
        matricule     :string;
        nPension      :string;
        nom           :string;
        prenom        :string;
        cin           :string;
        adresse       :string;
        ville         :string;
        codePostale   :string;
        dateNaissance :Date;
        sexe          :string;
        telephone     :string;
        telephone2    :string;
        rib           :string;
        rang          :number;
        situationVital:number;
        typeAffilie   :string;
        numAdherent   :string;
        email         :string;
         
        Qps            :IQp[];
        Cumules        :ICumuleQp[];
        QpMois         :IQpMois[];
        AvanceCheques  :IAvanceCheques[];
        misAjours      :IMisAjours[];
        cartes          :ICarte[];
        conjoints      :IConjoint[];
        enfants         :IEnfant[];
        
        somFreEngageTP :number;
        somRemMpscTP   :Number;
        somRemCnopsTP  :Number;
        somRemTP       :Number;
        nbrDossierT    :Number;
        somFreEngageDI :Number;
        somRemMpscDI   :Number;
        somRemCnopsDI  :Number;
        somRemDI       :Number;
        nbrDossierD    :Number;
        somFreEngage   :Number;
        somRemMpsc     :Number;
        somRemCnops    :Number;
        somRem         :Number;
        nbrDossier     :Number;
        
        somecumule     :Number;
        someAvanceMpsc :Number;

        totalRetenues  :Number;
        rest           :Number;
        reliquat       :Number;
    }
  
  
  export interface IQp {

    nDossier          :number;
    datePaiementTech  :Date;
    datePaiementReel  :Date;
    fraisEngage       :number;
    rembAmo           :number;
    rembMpsc          :number;
    totalRemb         :number;
    observation       :string;
    codeEchance       :number;
  }
  export interface ICumuleQp {

    id :number;
    cin :string;
    montant :number;
    observation :string;
    idUser :string;
    dateAffectation :Date;
  }
  export interface IQpMois {

    idPai :number;
    cinQp :string;
    date :Date;
    qp :number;
    rcar :string;
    typePai :string;
    codeEchanceQpMois :number;
    code303 :number;
    code304 :number;
    idUser :string;
    dateSaisie :Date;
    observation :string;
    cinPaiement :string;

  }

  export interface IAvanceCheques {
   idAvanceCheque :number;
   cinAv :string;
   montantAv :number;
   dateAvance :Date;
   userAvance :string;
   dateSaisie :Date;
   obser :string;
  }

  export interface ICarte
  {
    id            :number;
    cinAff        :string;
    dateArrive    :Date;
    dateEnvoie    :Date;
    nomPoretur    :string;
    cinPorteur    :string;
    disponible    :boolean;
    userReception :string;
    userEnvoie    :string;
  }

  export interface IEnfant
  {

    id :string;
    nom:string;
    prenom:string;
    cin:string;
    cinAff:string;
    dateNaissance:Date;
    sexe:string;
    observation :string;
    rang :number;
  }

  export interface IConjoint
  {

    nom :string;
    prenom :string;
    cin :string;
    cinAff :string;
    dateNaissance :Date;
    sexe :string;
    observation :string;
    rang :number;
  }

  /*export interface ITypeMisAjour {
    id :number;
    intitule :string;
   }

   export interface ILotMisAjour {
    Id          :number;
    DateDebut   :Date;
    UserDebut   :string;
    DateEnvoie  :Date;
  }*/


   export interface IMisAjours {
  
        id              :number;
        cinAffilie      :string;
        dateMaj         :Date;
        typeMaj         :string;
        info            :string;
        userMaj         :string;
        numLotMaj       :number;
        avcSCarte       :boolean;
        numCarte        :number;
        enCours         :boolean;
        infoIdentifiant :string;
        rejetMajs       :IRejetMaj[];
   }

   export interface IRejetMaj {
    id              :number;
    motif            :string;
    boite           :number;
    dateRejet       :Date;
    observation     :string;
    userRej        :string;
    idMaj          :number;
    corrigeRejets  :ICorrigeRejet[]

  }

  export interface ICorrigeRejet {
  id               :string;
  idRej            :string;
  userCorrigeRejet :string;
  dateCorrige      :string;
  numLot           :string;
  }